src/TenancyMailManager.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Stancl\Tenancy; // todo new Overrides namespace?
+
+use Illuminate\Mail\MailManager;
+use Stancl\Tenancy\Bootstrappers\MailTenancyBootstrapper;
+
+/**
+ * This class (paired with the MailTenancyBootstrapper)
+ * allows tenants to have their custom mailer credentials.
+ *
+ * Tenancy swaps Laravel's MailManager singleton for an instance of this class,
+ * which overrides the manager's get method to always resolve
+ * the mailers specified in the static $tenantMailers property
+ * instead of getting them from the $mailers property where they're cached.
+ *
+ * This is mainly used to solve the issue where
+ * mail gets sent with the incorrect (old) mailer credentials
+ * due to the mailer being cached in the manager's $mailers property
+ * and not getting updated when the tenant's credentials change.
+ *
+ * @see MailTenancyBootstrapper
+ */
+class TenancyMailManager extends MailManager
+{
+    /**
+     * Mailers to always resolve from the container (even when they're
+     * cached and available in the $mailers property).
+     */
+    public static array $tenantMailers = [
+        'smtp',
+    ];
+
+    /**
+     * Override the get method so that the mailers in $tenantMailers
+     * always get resolved, even when they're cached and available in the $mailers property
+     * so that the mailers have up-to-date mailer credentials.
```

Comment:
This feels a bit off when you read it in full

![CleanShot 2022-11-01 at 8 30 49@2x](https://user-images.githubusercontent.com/33033094/199321968-c91c095c-8fb8-48ab-9970-fdcff7407554.png)

The last line can probably be removed entirely